dbt/adapters/spark/python_submissions.py

```diff
@@ -7,78 +7,43 @@
 import dbt.exceptions
 from dbt.adapters.base import PythonJobHelper
 from dbt.adapters.spark import SparkCredentials
+from dbt.adapters.spark import __version__
 
-DEFAULT_POLLING_INTERVAL = 5
+DEFAULT_POLLING_INTERVAL = 10
 SUBMISSION_LANGUAGE = "python"
 DEFAULT_TIMEOUT = 60 * 60 * 24
+DBT_SPARK_VERSION = __version__.version
 
 
 class BaseDatabricksHelper(PythonJobHelper):
     def __init__(self, parsed_model: Dict, credentials: SparkCredentials) -> None:
-        self.check_credentials(credentials)
         self.credentials = credentials
         self.identifier = parsed_model["alias"]
-        self.schema = getattr(parsed_model, "schema", self.credentials.schema)
+        self.schema = parsed_model["schema"]
         self.parsed_model = parsed_model
         self.timeout = self.get_timeout()
         self.polling_interval = DEFAULT_POLLING_INTERVAL
+        self.check_credentials()
+        self.auth_header = {
+            "Authorization": f"Bearer {self.credentials.token}",
+            "User-Agent": f"dbt-labs-dbt-spark/{DBT_SPARK_VERSION} (Databricks)",
+        }
+
+    @property
+    def cluster_id(self) -> str:
+        return self.parsed_model.get("cluster_id", self.credentials.cluster_id)
```

Comment:
@ChenyuLInx I'm updating `dbt-databricks` and testing and found out that this is a mistake.

I think this should be:

```py
self.parsed_model["config"].get("cluster_id", self.credentials.cluster_id)
```